packages/react-workers/src/workers/viewportFeatures.worker.js

```diff
@@ -144,8 +148,34 @@ function getTimeSeries({ filters, column, stepSize, operation, operationColumn }
   postMessage({ result });
 }
 
-function getFilteredFeatures(filters) {
-  return !Object.keys(currentViewportFeatures).length
-    ? currentViewportFeatures
-    : currentViewportFeatures.filter(_buildFeatureFilter({ filters }));
+// See sorting details in utils/sorting.js
+function getRawFeatures({
+  filters,
+  limit = 10,
+  page = 1,
+  sortBy = [],
+  sortByDirection = 'asc'
+}) {
+  let data = [];
+  let numberPages = 0;
+
+  if (currentViewportFeatures) {
+    const features = applySorting(getFilteredFeatures(filters), {
+      sortBy,
+      sortByDirection
+    });
+
+    if (limit) {
+      numberPages = Math.ceil(features.length / limit);
+      data = features.slice(limit * Math.max(0, page - 1), limit * Math.max(1, page));
+    } else {
+      data = features;
+    }
+  }
+
+  postMessage({ result: { data, currentPage: page, pages: numberPages } });
+}
+
+function getFilteredFeatures(filters = {}) {
+  return _applyFilters(currentViewportFeatures, filters);
```

Comment:
I would check here if features need filters or not, Is there any reason why you have move it to `filters.js`?

Reply:
I think it's better to have that logic focused in one wrapper that can be used in many other parts that have the Object.keys(filters).length > 0 in every corner of the library or app.